src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -12253,6 +12253,41 @@ void M()
         }
 
         [Fact]
+        public void TupleWithDynamicInSeparateCompilations()
+        {
+            var lib_cs = @"
+public class C
+{
+    public static (dynamic, dynamic) M((dynamic, dynamic) x)
+    {
+        return x;
+    }
+}
+";
+
+            var source = @"
+class D
+{
+    static void Main()
+    {
+        var x = C.M((1, ""hello""));
+        x.Item1.DynamicMethod1();
+        x.Item2.DynamicMethod2();
+    }
+}
+";
+
+            var libComp = CreateCompilationWithMscorlib(lib_cs, assemblyName: "lib", references: new[] { ValueTupleRef, SystemRuntimeFacadeRef, SystemCoreRef });
+            libComp.VerifyDiagnostics();
+
+            var comp1 = CreateCompilationWithMscorlib(source, references: new[] { libComp.ToMetadataReference(), ValueTupleRef, SystemRuntimeFacadeRef });
+            comp1.VerifyDiagnostics();
+
+            var comp2 = CreateCompilationWithMscorlib(source, references: new[] { libComp.EmitToImageReference(), ValueTupleRef, SystemRuntimeFacadeRef });
+            comp2.VerifyDiagnostics();
+        }
+
```

Comment:
We need to test dynamic conversions as well.

1)   dynamic d = 1;   (int x, int y) t = (d, d);     // implicit from dynamic
2)   (dynamic x, dynamic y) t1 = (1, 1);            // implicit to dynamic 
3)   t1 = t  ; t1 = t;                                            // implicit to/from dynamic _type_ (should not compile)        
4)   t1 = ((dynamic, dynamic)) t;                       // explicit dynamic
5)   t = ((int, int)) t1;                                          // explicit dynamic

in all cases need to check that it does work as expected.
Could throw in a scenario that compiles and throws in runtime - i.e. put "new Exception()" in a dynamic and try convert to "string"